src/run_program.rs

```diff
@@ -279,7 +279,15 @@ impl<'a, D: Dialect> RunProgramContext<'a, D> {
         // put a bunch of ops on op_stack
         let SExp::Pair(op_node, op_list) = self.allocator.sexp(program) else {
             // the program is just a bitfield path through the env tree
-            let r: Reduction = traverse_path(self.allocator, self.allocator.atom(program), env)?;
+            let r: Reduction = self.allocator.visit_node(program, |node| -> Response {
+                match node {
+                    NodeVisitor::Buffer(buf) => traverse_path(self.allocator, buf, env),
+                    NodeVisitor::U32(val) => traverse_path_fast(self.allocator, *val, env),
+                    NodeVisitor::Pair(_, _) => {
+                        panic!("expected atom, got pair");
```

Comment:
Is this new behaviour? Shouldn't it be an `EvalError` rather than a panic?

Reply:
it's not new. Previously we called `self.allocator.atom(program)` which panics if the node is not an atom.

Reply:
I see, the "it's not a pair" check has already been done at this point.

Reply:
well, in the new version, we call `visit_node()` instead of `atom()`. so this is really the counterpart to the one in `atom()`